internal/cli/kraft/lib/create/create.go

```diff
@@ -0,0 +1,315 @@
+	"kraftkit.sh/unikraft/lib/template"
+)
+
+type CreateOptions struct {
+	ProjectName     string `long:"project-name" usage:"Set the project name to the template"`
+	LibraryName     string `long:"library-name" usage:"Set the library name to the template"`
+	LibraryKName    string `long:"library-kname" usage:"Set the library kname to the template"`
+	Version         string `long:"version" short:"v" usage:"Set the library version to the template"`
+	Description     string `long:"description"  usage:"Set the description to the template"`
+	AuthorName      string `long:"author-name" usage:"Set the author name to the template"`
+	AuthorEmail     string `long:"author-email" usage:"Set the author email to the template"`
+	InitialBranch   string `long:"initial-branch" usage:"Set the initial branch name to the template"`
+	CopyrightHolder string `long:"copyright-holder" usage:"Set the copyright holder name to the template"`
+	Origin          string `long:"origin" usage:"Source code origin URL."`
+	NoProvideMain   bool   `long:"no-provide-main" usage:"Do not provide provide-main to the template"`
+	GitInit         bool   `long:"git-init" usage:"Init git through the creating library"`
+	WithPatchedir   bool   `long:"patch-dir" usage:"provide patch directory to the template"`
+	SoftPack        bool   `long:"soft-pack" usage:"Softly pack the component so that it is available via kraft list"`
+	ProjectPath     string `long:"project-path" usage:"Where to create library"`
+}
+
+// Create creates a library template.
+func Create(ctx context.Context, opts *CreateOptions, args ...string) error {
+	if opts == nil {
+		opts = &CreateOptions{}
+	}
+
+	return opts.Run(ctx, args)
+}
+
+func NewCmd() *cobra.Command {
+	cmd, err := cmdfactory.New(&CreateOptions{}, cobra.Command{
+		Short:   "Initialise a library template",
+		Use:     "create [FLAGS] [NAME]",
+		Aliases: []string{"init"},
+		Long: heredoc.Doc(`
+		Creates a library template
+		`),
+		Example: heredoc.Doc(`
+			$ kraft lib create
+			$ kraft lib create sample-project
+			`),
+		Annotations: map[string]string{
+			cmdfactory.AnnotationHelpGroup: "lib",
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	return cmd
+}
+
+func (*CreateOptions) Pre(cmd *cobra.Command, _ []string) error {
+	ctx, err := packmanager.WithDefaultUmbrellaManagerInContext(cmd.Context())
+	if err != nil {
+		return err
+	}
+	cmd.SetContext(ctx)
+	return nil
+}
+
+func (opts *CreateOptions) Run(ctx context.Context, args []string) error {
+	var err error
+
+	if len(args) > 0 {
+		opts.ProjectName = args[0]
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	if !config.G[config.KraftKit](ctx).NoPrompt {
+		if !opts.GitInit {
+			input := textinput.New("Do you want to intialise library with git [Y/n]:")
+			input.Placeholder = "Enter Y for yes"
+			inputValue, err := input.RunPrompt()
+			if err != nil {
+				return err
+			}
+			if inputValue == "Y" || inputValue == "y" {
+				opts.GitInit = true
+			}
```

Comment:
For everything prompt related there are now tools inside kraftkit that can handle those, for example you can use `selection.Select` to make users select between `yes` and `no` (see https://github.com/unikraft/kraftkit/blob/staging/tui/selection/select.go )

For the ones that requires users to specify a name, you will have to create another package `specify` or `prompt` or something else, which waits for a user given string

Reply:
Made a new package `specify` 👍🏽